### DISCUSSION TITLE
Firmware upgrade feature

### DISCUSSION THREAD

**Discussion by kaklik** [Open-Gamma-Project/Gamma-MCA#27] [Ideas]
The currently used devices that should be possibly connected to Gamma MCA, use bootloaders (more specifically Arduino bootloaders). 
That situation reveals the possible feature which allows an automatic firmware update in the connected devices.

The limitation seems to be a lack of an existing Arduino bootloader interface in javascript. 
I found this repository: 

- [arduino-web-uploader](https://github.com/dbuezas/arduino-web-uploader)

But it seems it currently relies on WebUSBSerial support.

**Comment by NuclearPhoenixx** [comment]
Not a fan, I think this is not a good idea as a general feature. Firmware uploading would open up a pandoras box of edge cases, compatability problems and endless device checking.

This would only work for Arduino-compatible boards and even then it is restricted to only a few different bootloaders it looks like. IMO this is better suited with an online IDE like https://www.espruino.com/ide/ or Arduino Create, but not here.

**Comment by NuclearPhoenixx** [comment]
Closing this discussion too since it's not really practical to implement due to the sheer number of different devices. Feel free to add your opinion.
